Document CircleCI recommendations

Peter-Darton-i2 opened the pull request:
For: #37 

As described in #37, the documentation has a hole where the CircleCI recommendations should be.

CircleCI support are recommending this logger to CircleCI users (like me), so it's time we plugged that hole in the documentation.

**Note:** No functional changes in here, just documentation.

Peter-Darton-i2 committed:
Write CircleCI recommendations

Peter-Darton-i2 committed:
Minor corrections/tidy-up

Peter-Darton-i2 committed:
Minor corrections/tidy-up

Peter-Darton-i2 committed:
Recommend FailureBodyFormat for CircleCI

Siphonophora commented on docs/circleci-recommendation.md:
```diff
@@ -1,3 +1,53 @@
+
+- `Default`:
+  ![MethodFormat Default](assets/circleci-test-collapsed-with-methodname-default.png)
+- `Class`:
```
Go ahead and combine the class and full options. Also, add extra line breaks so the images are on the next line.

Siphonophora commented on docs/circleci-recommendation.md:
````diff
@@ -1,3 +1,53 @@
+```yml
+- run:
+   command: |
+     vstest.console.exe ... --logger:"junit;LogFilePath=TestResults/test-result.xml;FailureBodyFormat=Verbose"
````
our examples all include `dotnet test` instead vstest.console. Unless the console is mandatory for some reason, lets align it with the other examples.

Siphonophora commented on docs/circleci-recommendation.md:
```diff
@@ -1,3 +1,53 @@
+Maintenance note:
+Findings as of July 20 2023.
+ - Only data in a <testcase> element is shown.
+ - The logger only puts console text in at the <testsuite> level so CircleCI ignores it.
```
I'm a bit confused here. It says below system-out and system-err are shown, but this sounds like it isn't shown?

Why don't we consolidate this so its all user visible. Go ahead and be explicit about limitations in the docs.

Peter-Darton-i2 replied:
Re: confusion

Right now, the logger takes all the console output from both the setup, closedown and individual tests, turns it into a single string, and places that in a <system-out> element in the test**suite** ... but, at present, CircleCI ignores anything in the test**suite**; CircleCI only shows data from the test**case**, not the test**suite**.

e.g.
```xml
<testsuites>
  <testsuite>
    <testcase>
      <properties>
        <property name="CircleCI" value="Does not show these" />
      </properties>
      <failure message="This scan be seen in CircleCI">So can this</failure>
      <system-out>This would be shown too, but the logger does not output this.<system-out>
      <system-err>I expect the same would apply for system-err too</system-err>
    </testcase>
    <system-out>The logger puts data here but CircleCI ignores it</system-out>
    <system-err>I expect the same would apply for system-err too</system-err>
  </testsuite>
</testsuites>
```


Re: Why
CircleCI is a moving target; I thought it'd be helpful to include some kind of "this was the case at \<date>" text so that the next person to edit this file could better understand why it said what I've written.
... but, while I like to have some kind of comment saying why things have been written the way they were, I don't believe that having this "out in the open" yields the prettiest documentation.

It could be turned into some kind of "footnote" if you'd prefer that kind of commentary visible for all to see.

Peter-Darton-i2 replied:
As it needed explanation (IMO, if someone asks questions about documentation then the documentation wasn't clear enough) I've moved the "maintenance note" into a footnote _and_ added a bit of sample XML highlighting what CircleCI does/does-not display.

Siphonophora commented:
@Peter-Darton-i2 Thanks for writing this up. Much appreciated!

Peter-Darton-i2 committed:
Address review comments

Peter-Darton-i2 committed:
Move maintenance comment into a footnote

Peter-Darton-i2 committed:
Incorporate suggestions from CircleCI

Peter-Darton-i2 commented:
I've made a few changes; I think I've addressed all the previous review comments, and incorporated some changes CircleCI requested too.

I think this is ready for merge.

Siphonophora commented:
@Peter-Darton-i2 Thanks for the contribution
